django/db/models/aggregates.py

```diff
@@ -211,3 +215,47 @@ def __init__(self, expression, sample=False, **extra):
 
     def _get_repr_options(self):
         return {**super()._get_repr_options(), "sample": self.function == "VAR_SAMP"}
+
+
+class JSONArrayAgg(Aggregate):
+    function = "JSON_ARRAYAGG"
+    output_field = JSONField()
+    arity = 1
+
+    def as_sql(self, compiler, connection, **extra_context):
+        if not connection.features.supports_aggregate_filter_clause:
```

Comment:
You should only raise if there's an actual filter; otherwise it's actually supported?

```suggestion
        if self.filter and not connection.features.supports_aggregate_filter_clause:
```